service/src/main/kotlin/fi/espoo/evaka/shared/async/AsyncJobRunner.kt

```diff
@@ -118,6 +118,11 @@ class AsyncJobRunner<T : AsyncJobPayload>(
         tx.afterCommit(wakeUpHook)
     }
 
+    fun plan(tx: Database.Transaction, jobs: List<JobParams<*>>) {
```

Comment:
This should use the `T` type parameter instead of a wildcard: `JobParams<T>`
The wildcard accepts *any* type of async job, even jobs that this particular async job runner can't execute. In this kind of scenario the insert to database succeeds (because it doesn't care) but the wake up hook breaks...not a critical problem since the jobs will eventually be executed by the correct runner, but this can cause unexpected delays and possible test flakiness.